macOS/DuckDuckGo/AIChat/AIChatTabOpener.swift

```diff
@@ -19,97 +19,125 @@
 import Foundation
 import AIChat
 
-protocol AIChatTabOpening {
-    @MainActor
-    func openAIChatTab(_ query: String?, with linkOpenBehavior: LinkOpenBehavior)
-
-    @MainActor
-    func openAIChatTab(_ value: AddressBarTextField.Value, with linkOpenBehavior: LinkOpenBehavior)
-
-    @MainActor
-    func openNewAIChatTab(_ aiChatURL: URL, with linkOpenBehavior: LinkOpenBehavior)
-
-    @MainActor
-    func openNewAIChatTab(withPayload payload: AIChatPayload)
+enum AIChatContent {
+    case newChat
+    case query(String?, shouldAutoSubmit: Bool = true)
+    case url(URL)
+    case payload(AIChatPayload)
+    case restoration(AIChatRestorationData)
+}
 
+protocol AIChatTabOpening {
     @MainActor
-    func openNewAIChatTab(withChatRestorationData data: AIChatRestorationData)
-}
+    func openAIChatTab(with content: AIChatContent, behavior: LinkOpenBehavior)
 
-extension AIChatTabOpening {
     @MainActor
-    func openAIChatTab() {
-        openAIChatTab(nil, with: .currentTab)
-    }
+    func openNewAIChat(in linkOpenBehavior: LinkOpenBehavior)
 }
 
 struct AIChatTabOpener: AIChatTabOpening {
     private let promptHandler: AIChatPromptHandler
-    private let addressBarQueryExtractor: AIChatAddressBarPromptExtractor
-    private let windowControllersManager: WindowControllersManagerProtocol
+    private let aiChatTabManaging: AIChatTabManaging
 
     let aiChatRemoteSettings = AIChatRemoteSettings()
 
     init(
         promptHandler: AIChatPromptHandler,
-        addressBarQueryExtractor: AIChatAddressBarPromptExtractor,
-        windowControllersManager: WindowControllersManagerProtocol
+        aiChatTabManaging: AIChatTabManaging
     ) {
         self.promptHandler = promptHandler
-        self.addressBarQueryExtractor = addressBarQueryExtractor
-        self.windowControllersManager = windowControllersManager
+        self.aiChatTabManaging = aiChatTabManaging
     }
 
+    // MARK: - New Simplified API
+
     @MainActor
-    func openAIChatTab(_ value: AddressBarTextField.Value, with linkOpenBehavior: LinkOpenBehavior) {
-        let query = addressBarQueryExtractor.queryForValue(value)
-
-        // We don't want to auto-submit if the user is opening duck.ai from the SERP
-        // https://app.asana.com/1/137249556945/project/1204167627774280/task/1210024262385459?focus=true
-        let shouldAutoSubmit: Bool
-        if case let .url(_, url, _) = value {
-            shouldAutoSubmit = !url.isDuckDuckGoSearch
-        } else {
-            shouldAutoSubmit = true
```

Comment:
This logic has been extracted level up to [AIChatAddressBarPromptExtractor.swift](https://github.com/duckduckgo/apple-browsers/pull/1982/files#diff-738a67a616b44c3af08f191a469473c058202a924a4783bb32b1b48749ca43e2) which is called ahead of `openAIChatTab` to obtain the content